### PR TITLE
#770 auth api

### LINKED ISSUE #770: Authentication: add API
Some functions to begin with:
- [x] fetch list of stores
- [x] handle errors when fetching?
- [x] send login request
- [x] display login error

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #770 

Adds the api and queries, shows login errors - generic if there was an authentication error since telling people that they have the wrong username is a helpful attack vector. as it is, the API thinks for a while if you have the right username and wrong password, but responds immediately if the username is wrong so it's pretty obvious.

Still need to store the token and other login details for display in the site footer. 
Will need to add refresh support, add the token (and `storeId` apparently) to queries. and add mocks for store and token.
Also want to remember last login username and store.

### CODE REVIEW

File: packages/common/src/hooks/useHostContext.ts [hunk 1]
```diff
@@ -46,7 +46,7 @@ export const useHostContext = create<HostContext>(set => ({
   detailPanelRef: null,
 
   setStore: store => set(state => ({ ...state, store })),
-  store: { id: '4321dcba', name: 'Central Warehouse' },
+  store: { id: '4321dcba', code: 'Central Warehouse' },
```
Comment (mark-prins) [inline packages/common/src/hooks/useHostContext.ts:49]: the API doesn't provide name or description, I've created an issue to include those.

File: packages/common/src/ui/components/buttons/standard/BaseButton.tsx [hunk 2]
```diff
@@ -4,30 +4,29 @@ import {
   Button as MuiButton,
   ButtonProps as MuiButtonProps,
   styled,
+  Theme,
 } from '@mui/material';
 
+const translateColor = (theme: Theme, color?: string) => {
```
Comment (mark-prins) [inline packages/common/src/ui/components/buttons/standard/BaseButton.tsx:10]: this was a little offroad - the primary button was showing secondary colours on hover and I didn't like it

File: packages/host/src/Host.tsx [hunk 3]
```diff
@@ -30,30 +30,32 @@ const queryClient = new QueryClient({
 });
 
 const Host: FC = () => (
-  <React.Suspense fallback={<RandomLoader />}>
+  <React.Suspense fallback={<div />}>
```
Comment (mark-prins) [inline packages/host/src/Host.tsx:33]: oh. heh. The loading spinners were suspending and had no fallback of their own 🙄 
I don't know if this is kosher or not, seems to work 🤷

File: packages/host/src/components/Login/Login.tsx [hunk 4]
```diff
@@ -27,18 +33,76 @@ const StoreAutocompleteInput: React.FC<AutocompleteRenderInputParams> =
     );
   };
 
+const storeSorter = (a: Store, b: Store) => {
```
Comment (mark-prins) [inline packages/host/src/components/Login/Login.tsx:36]: the api provides stores in a random ( database insertion? ) order and has no sort option.
I like alphabetical.. so have done this for now and raised an issue for the api team.

File: packages/host/src/components/Login/Login.tsx [hunk 5]
```diff
@@ -87,30 +151,64 @@ export const Login: React.FC = ({}) => {
...
+                }}
+              />
+              <LoginTextInput
+                fullWidth
+                label={t('heading.password')}
+                type="password"
+                value={password}
+                onChange={e => setPassword(e.target.value)}
+                inputProps={{
+                  autoComplete: 'current-password',
```
Comment (mark-prins) [inline packages/host/src/components/Login/Login.tsx:175]: I dunno.. chrome said this was important 🤷  same as adding the `<form/>` - that helps the password savers to work

**Review by joshxg — APPROVED** [review]
Ah crap I didn't submit sorry 😭

**Pull request merged**